packages/web-components/src/components/countdown/countdown.ts

```diff
@@ -0,0 +1,229 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2024
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { html, property, state, LitElement } from 'lit-element';
+import { carbonElement as customElement } from '@carbon/web-components/es/globals/decorators/carbon-element';
+import settings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings.js';
+import { LocaleAPI } from '@carbon/ibmdotcom-services/es/services/Locale/index';
+import MediaQueryMixin, {
+  MQBreakpoints,
+  MQDirs,
+} from '@carbon/ibmdotcom-web-components/es/component-mixins/media-query/media-query';
+
+const { stablePrefix: c4dPrefix } = settings;
+
+const ms_per = {
+  second: 1_000,
+  minute: 1_000 * 60,
+  hour: 1_000 * 60 * 60,
+  day: 1_000 * 60 * 60 * 24,
+};
+
+const units = Object.keys(ms_per);
+
+const getFormatters = (locale: Locale, labelType: UnitDisplay) => {
+  const lc_cc = `${locale.lc}-${locale.cc}`;
+
+  return Object.fromEntries(
+    units.map((unit) => [
+      `to_${unit}s`,
+      new Intl.NumberFormat(lc_cc, {
+        style: 'unit',
+        unit,
+        unitDisplay: labelType,
```

Comment:
Typescript is not happy here. I believe it's b/c we may have a value of `none`, which is [not allowed for this option](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#unitdisplay). I think if we adjust the `UnitDisplay` type on line 72 to append on `| undefined`, and then add the following if statement below line 31, we should be good:

```
// The Typescript compiler sees unitDisplay as type
// "short" | "long" | "narrow" | undefined. Convert 'none' to undefined to
// avoid passing an unknown value to the unitDisplay option.
if (labelType === 'none') {
  labelType = undefined;
}
```